java/cdk/src/main/java/sleeper/cdk/stack/TransactionLogSnapshotStack.java

```diff
@@ -131,7 +145,73 @@ public TransactionLogSnapshotCreationStack(
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "transaction-log-snapshot-deletion"));
+        IFunction snapshotDeletionTrigger = statestoreJar.buildFunction(this, "TransactionLogSnapshotDeletionTrigger", builder -> builder
+                .functionName(triggerFunctionName)
+                .description("Creates batches of Sleeper tables to delete old transaction log snapshots for and puts them on a queue to be processed")
+                .runtime(Runtime.JAVA_11)
+                .handler("sleeper.statestore.TransactionLogSnapshotDeletionTriggerLambda::handleRequest")
+                .environment(Utils.createDefaultEnvironment(instanceProperties))
+                .memorySize(instanceProperties.getInt(TABLE_BATCHING_LAMBDAS_MEMORY_IN_MB))
+                .timeout(Duration.seconds(instanceProperties.getInt(TABLE_BATCHING_LAMBDAS_TIMEOUT_IN_SECONDS)))
+                .logGroup(createLambdaLogGroup(this, "TransactionLogSnapshotDeletionTriggerLogGroup", triggerFunctionName, instanceProperties)));
+        IFunction snapshotDeletionLambda = statestoreJar.buildFunction(this, "TransactionLogSnapshotDeletion", builder -> builder
+                .functionName(deletionFunctionName)
+                .description("Deletes old transaction log snapshots for tables")
+                .runtime(Runtime.JAVA_11)
+                .handler("sleeper.statestore.TransactionLogSnapshotDeletionLambda::handleRequest")
+                .environment(Utils.createDefaultEnvironment(instanceProperties))
+                .memorySize(1024)
+                .timeout(Duration.minutes(1))
+                .logGroup(createLambdaLogGroup(this, "TransactionLogSnapshotDeletionLogGroup", deletionFunctionName, instanceProperties)));
+
+        Rule rule = Rule.Builder.create(this, "TransactionLogSnapshotDeletionSchedule")
+                .ruleName(SleeperScheduleRule.TRANSACTION_LOG_SNAPSHOT_DELETION.buildRuleName(instanceProperties))
+                .schedule(Schedule.rate(Duration.minutes(
+                        instanceProperties.getLong(TRANSACTION_LOG_SNAPSHOT_DELETION_LAMBDA_PERIOD_IN_MINUTES))))
+                .targets(List.of(new LambdaFunction(snapshotDeletionTrigger)))
+                .enabled(!shouldDeployPaused(this))
+                .build();
+        instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_DELETION_RULE, rule.getRuleName());
+
+        Queue deadLetterQueue = Queue.Builder
+                .create(this, "TransactionLogSnapshotDeletionDeadLetterQueue")
+                .queueName(String.join("-", "sleeper", instanceProperties.get(ID), "TransactionLogSnapshotDeletionDLQ.fifo"))
+                .fifo(true)
+                .build();
+        Queue queue = Queue.Builder
+                .create(this, "TransactionLogSnapshotDeletionQueue")
+                .queueName(String.join("-", "sleeper", instanceProperties.get(ID), "TransactionLogSnapshotDeletionQ.fifo"))
```

Comment:
These queue names are over the length limit for AWS resource names.

Reply:
Never mind, this is within the limit for SQS. I thought all resources were limited at 64 characters.